airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/source.py

```diff
@@ -140,6 +141,16 @@ def _make_authenticator(config: Mapping[str, Any]):
 
     @staticmethod
     def _choose_profiles(config: Mapping[str, Any], profiles: List[Profile]):
-        if not config.get("profiles"):
+        filtered_profiles = []
+        if config.get("profiles"):
+            for profile in list(filter(lambda profile: profile.profileId in config["profiles"], profiles)):
+                filtered_profiles.append(profile)
+        if config.get("marketplace_ids"):
+            for profile in list(filter(lambda profile: profile.profileId in config["marketplace_ids"], profiles)):
```

Comment:
Here you need to filter based on `profile.accountInfo.marketPlaceId` not `profile.profileId`.

Reply:
Can you add a unit test for this function or check if the connector doesn't have one already?

Reply:
> Here you need to filter based on `profile.accountInfo.marketPlaceId` not `profile.profileId`.

Ah, good catch. I will fix that.